cordapp-example/workflows-kotlin/src/main/kotlin/com/example/flow/PaybackFlow.kt

```diff
@@ -29,6 +36,19 @@ import java.util.*
  * All methods called within the [FlowLogic] sub-class need to be annotated with the @Suspendable annotation.
  */
 object PaybackFlow {
+    fun findMoney(serviceHub: ServiceHub, party: Party): StateAndRef<CashState> {
+        // looking for money that should be printed on ExampleFlow
+        // this is not the best way, but I use this way to exercise, I will do better in module 4
```

Comment:
As an exercise, you can also use a custom query criteria using the custom schema that you created for the `CashState`, i.e. you can query against `owner == party`.   
Below is an example taken from this [page](https://docs.corda.net/docs/corda-os/4.4/api-vault-query.html):
```java
QueryCriteria generalCriteria = new VaultQueryCriteria(Vault.StateStatus.ALL);

FieldInfo attributeCurrency = getField("currency", CashSchemaV1.PersistentCashState.class);
FieldInfo attributeQuantity = getField("pennies", CashSchemaV1.PersistentCashState.class);

CriteriaExpression currencyIndex = Builder.equal(attributeCurrency, "USD");
CriteriaExpression quantityIndex = Builder.greaterThanOrEqual(attributeQuantity, 10L);

QueryCriteria customCriteria2 = new VaultCustomQueryCriteria(quantityIndex);
QueryCriteria customCriteria1 = new VaultCustomQueryCriteria(currencyIndex);


QueryCriteria criteria = generalCriteria.and(customCriteria1).and(customCriteria2);
Vault.Page<ContractState> results = vaultService.queryBy(Cash.State.class, criteria);
```